packages/svelte/src/compiler/phases/3-transform/client/visitors/javascript-runes.js

```diff
@@ -27,35 +65,10 @@ export const javascript_visitors_runes = {
 				const is_private = type === 'PrivateIdentifier';
 				if (is_private) private_ids.push(name);
 
-				if (definition.value?.type === 'CallExpression') {
-					const rune = get_rune(definition.value, state.scope);
-					if (
-						rune === '$state' ||
-						rune === '$state.frozen' ||
-						rune === '$derived' ||
-						rune === '$derived.by'
-					) {
-						/** @type {import('../types.js').StateField} */
-						const field = {
-							kind:
-								rune === '$state'
-									? 'state'
-									: rune === '$state.frozen'
-										? 'frozen_state'
-										: rune === '$derived.by'
-											? 'derived_call'
-											: 'derived',
-							// @ts-expect-error this is set in the next pass
-							id: is_private ? definition.key : null
-						};
-
-						if (is_private) {
-							private_state.set(name, field);
-						} else {
-							public_state.set(name, field);
-						}
-					}
-				}
+				create_state_field(definition, is_private, name);
+			} else if (definition.type === 'PropertyDefinition' && definition.key.type === 'Literal') {
+				const name = definition.key.value?.toString().replace('-', '_');
+				if (name) create_state_field(definition, false, name);
```

Comment:
Why would the property necessarily be public?

I mean there is a [parsing error](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAAE2WPzWrDMAzHX0Xo1LIs665pEhh7hB3rHjJVZaaubSw5pQS_-_BSGGMnof-H-GnBs3Us2B0W9NOVscO3GLFBvce6yMxOGRuUkBNVpRdKNupovFFykwi8h-z1gymxuvtSdaMx2XlSBoNSDdLneXKZDcIAu73xayp4cpYuMMBmO4yPqlH9snL41zzC0wCva6bUUdYzFLwoUIWAATzf_gJttnvj-5dfat9_ZtXgYWnb9qdWHs9YukjNrv6IDV7DyZ4tn7DTlLkcyzcRnlaDMAEAAA==) 

but it's [valid ts syntax](https://www.typescriptlang.org/play/?#code/MYGwhgzhAEDCD2BXAdgFwMoFNgCdOpAE8BvAWACgBISgBxwEsA3MVTaAIgmz2FQFpmIRJnbQAvNAAMAbgoVq8ZKHrAA1uOgAKAJRiAfGSrVUAC3oQA2p27Z+g4ewC60ANQSAjPMoBfed6A)

Reply:
Just opened an [issue](https://github.com/sveltejs/svelte/issues/11332) about this

Reply:
It is a parsing error in JavaScript. If you add the `lang="TS"` it works but I have to check how it is compiled down because in pure JavaScript you can't have a literal private field

Reply:
mentioned this on the issue, `lang="ts"` is missing